src/ol/expr/cpu.js

```diff
@@ -256,7 +256,22 @@ function compileAccessorExpression(expression, context) {
   const name = /** @type {string} */ (nameExpression.value);
   switch (expression.operator) {
     case Ops.Get: {
-      return (context) => context.properties[name];
+      return (context) => {
+        const args = expression.args;
+        let value = context.properties[name];
+        for (let i = 1, ii = args.length; i < ii; ++i) {
+          if (value === null || typeof value !== 'object') {
+            break;
+          }
+          const keyExpression = /** @type {LiteralExpression} */ (args[i]);
+          const key = /** @type {string|number} */ (keyExpression.value);
+          if (typeof key !== 'number' && Array.isArray(value)) {
+            break;
```

Comment:
Looks like with properties `{"a": 1}` and `["get", "a", "b"]` it would currently return `1`?
Same for properties `{"a": [1]}` and `["get", "a", "b"]` would return `[1]`?

Should this return `undefined` instead?

Reply:
Returning `undefined` in such cases was my initial approach, but then I thought it might be useful to allow data structures with variable nesting depth - and be backwards compatible with the removed type hint as last argument.

If people think there is no practical use of this, and it rather adds confusion, I'll be happy to go back to the initial approach and make this a breaking change.

Reply:
I think we should move on from the type hint.

Reply:
Ok, updated. Now it's a breaking change.